docs/fundamentals/code-analysis/quality-rules/ca1418.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "CA1418: Use valid platform string (code analysis)"
+description: "Learn about code analysis rule CA1418: Use valid platform string"
+ms.date: 09/01/2020
+ms.topic: reference
+f1_keywords:
+  - "UseValidPlatformString"
+  - "CA1418"
+helpviewer_keywords:
+  - "UseValidPlatformString"
+  - "CA1418"
+author: buyaa-n
+ms.author: bunamnan
+---
+# CA1418: Validate platform compatibility
+
+|||
+|-|-|
+| **Rule ID** |CA1418|
+| **Category** |[Interoperability](interoperability-warnings.md)|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+Platform compatibility analyzer requires a valid platform name and version. Violations are reported if the platform string provided to tha the <xref:System.Runtime.Versioning.OSPlatformAttribute> constructor consists **unknown** platform name or if the optional version part is **invalid**.
+
+## Rule description
+
+The platform compatibility attributes derived from <xref:System.Runtime.Versioning.OSPlatformAttribute> use string literals for OS platform names with optional version part. The string should consist a **known** platform name and have no version or with a **valid** version part.
+
+The **known** platform names list is populated from two places:
+
+- `PlatformName` part of <xref:System.OperatingSystem> guard methods `OperatingSystem.Is(PlatformName)[VersionAtLeast]()`, for example guard method <xref:System.OperatingSystem.IsWindows?displayProperty=nameWithType> would make the `Windows` part of the known platforms.
+- The project's MSBuild item group of `SupportedPlatform`s, including the default [MSBuild SupportedPlatforms list](https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedPlatforms.props). This is the project specific knowledge of known platforms, this allows class library authors to add more platforms into the known platforms list:
+
+   ```XML
+    <ItemGroup>
+      <SupportedPlatform Include="PlatformName" />
+    </ItemGroup>
+  ```
+
+If the platform string contains **version** part it should be a **valid** <xref:System.Version> with a format: `major.minor[.build[.revision]]`.
+
+### Violations
+
+- `Solaris` is an **unknown** platform name because it is **not** included in the default [MSBuild SupportedPlatforms list](https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedPlatforms.props) and there is **no** guard method like `OperatingSystem.IsSolaris()`in <xref:System.OperatingSystem> class.
```

Comment:
```suggestion
- `Solaris` is an **unknown** platform name because it's not included in the default [MSBuild SupportedPlatforms list](https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.SupportedPlatforms.props) and there's no guard method named `OperatingSystem.IsSolaris()` in the <xref:System.OperatingSystem> class.
```